packages/grid/_modules_/grid/hooks/core/useGridApiInitialization.ts

```diff
@@ -60,7 +60,6 @@ export function useGridApiInitialization(
     return () => {
       logger.info('Unmounting Grid component. Clearing all events listeners.');
       api.publishEvent(GridEvents.unmount);
-      api.unstable_eventManager.removeAllListeners();
```

Comment:
Each component/hook should be responsible for unsubscribing only its own listeners. Removing all listeners at once could hide something.